core/src/main/java/com/linecorp/armeria/server/annotation/AdditionalHeader.java

```diff
@@ -16,12 +16,17 @@
 package com.linecorp.armeria.server.annotation;
 
 import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
  * Annotation for an additional HTTP header.
  */
-@Target(ElementType.ANNOTATION_TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(AdditionalHeaders.class)
+@Target({ElementType.TYPE, ElementType.METHOD, ElementType.ANNOTATION_TYPE})
 public @interface AdditionalHeader {
```

Comment:
How about changing the type of `value()` `String[]` instead of `String` so that a user can specify a header with multiple values?

```java
@AdditionalHeader(name = "foo", value = "bar")
@AdditionalHeader(name = "foo", value = "baz") // Does not work...
// vs.
@AdditionalHeader(name = "foo", value = { "bar", "baz" })
```